airflow/migrations/versions/6e96a59344a4_make_taskinstance_pool_not_nullable.py

```diff
@@ -0,0 +1,60 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Make TaskInstance.pool not nullable
+
+Revision ID: 6e96a59344a4
+Revises: 939bb1e647c8
+Create Date: 2019-06-13 21:51:32.878437
+
+"""
+
+from alembic import op
+import sqlalchemy as sa
+
+# revision identifiers, used by Alembic.
+revision = '6e96a59344a4'
+down_revision = '939bb1e647c8'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    """
+    Make TaskInstance.pool field not nullable.
+    """
+    # use batch_alter_table to support SQLite workaround
+    with op.batch_alter_table('task_instance') as batch_op:
+        batch_op.alter_column(
+            column_name='pool',
+            type_=sa.String(50),
+            nullable=False,
```

Comment:
Good catch. I think we might need to do bulk update here. For upgrade, we will need to change all the existing rows with `pool=None` to `pool='default_pool'` and vice versa for downgrade.

Reply:
added.

Reply:
For future reference (i.e. don't change it, it is fine as it is) : I think we can get away with just defining the PK and columns we want to alter, so

```python
class TaskInstance(Base):
    __tablename__ = "task_instance"

    task_id = Column(String(ID_LEN), primary_key=True)
    dag_id = Column(String(ID_LEN), primary_key=True)
    execution_date = Column(UtcDateTime, primary_key=True)
    pool = Column(String(50), nullable=False)
```

would work. I _think_.

Reply:
FYI, this will break on MSSQL which I known isn't supported atm. One would need to drop the index 'ti_pool', alter the column, and recreate the index.

Reply:
Doesn't that make that an expensive operation on MSSQL - needing to re-create the entire index?  (Not that there's an option but that seems expensive)

Reply:
You're right, there is no other way: https://docs.microsoft.com/en-us/sql/t-sql/statements/alter-table-transact-sql?view=sql-server-2017. It's a costly thing, but whether it takes a long time or not depends on the size of the table and your hardware. We are using MSSQL as a backend for Airflow and rebuilding the indexes is the least of my concern :-)